packages/application-config/src/process-config.ts

```diff
@@ -28,21 +40,182 @@ type ProcessConfigOptions = Partial<LoadingConfigOptions> & {
+    }),
+    oAuthScopes: appConfig.oAuthScopes,
+    additionalOAuthScopes: appConfig?.additionalOAuthScopes,
+  });
+
+  if (isCustomViewData(configurationData)) {
+    const hostUriPath = (appConfig as JSONSchemaForCustomViewConfigurationFiles)
+      .env.development.hostUriPath;
+    const defaultHostUriPath = oidcConfig.initialProjectKey
+      ? `/${oidcConfig.initialProjectKey}/${entryPointUriPath}`
+      : `/${entryPointUriPath}`;
+    const hostUrl = new URL(
+      hostUriPath || defaultHostUriPath,
+      'http://localhost:3001'
+    );
+    return omitDevConfigIfEmpty<ApplicationRuntimeEnvironmentForDevelopment>({
+      oidc: oidcConfig,
+      customViewConfig: configurationData,
+      customViewHostUrl: hostUrl.href,
+    });
+  }
+
+  return omitDevConfigIfEmpty<ApplicationRuntimeEnvironmentForDevelopment>({
+    oidc: oidcConfig,
+    menuLinks: {
+      icon: configurationData.icon,
+      ...configurationData.mainMenuLink,
+      submenuLinks: configurationData.submenuLinks,
+    },
+    // @ts-expect-error: the `accountLinks` is not explicitly typed as it's only used by the account app.
+    accountLinks: configurationData.accountLinks,
+  });
+};
+
+const getRuntimeEnvironmentConfig = ({
+  isProd,
+  configurationData,
+  additionalAppEnv,
+  mcApiUrl,
+  cdnUrl,
+  appUrl,
+  appEnvKey,
+  revision,
+  appConfig,
+}: {
+  isProd: boolean;
+  configurationData: CustomApplicationData | CustomViewData;
+  additionalAppEnv: Record<string, unknown>;
+  mcApiUrl: URL;
+  cdnUrl: URL;
+  appUrl: URL;
+  appEnvKey: string;
+  revision: string;
+  appConfig:
+    | JSONSchemaForCustomApplicationConfigurationFiles
+    | JSONSchemaForCustomViewConfigurationFiles;
+}): ApplicationRuntimeEnvironment => {
+  const entryPointUriPath = isCustomViewData(configurationData)
+    ? // When the application acts as the host for Custom Views, there is no real
+      // entry point to be used, therefore we use a special identifier.
+      CUSTOM_VIEW_HOST_ENTRY_POINT_URI_PATH
+    : configurationData.entryPointUriPath;
+
+  // The real application ID is only used in production.
+  // In development, we prefix the entry point with the "__local" prefix.
+  // This is important to determine to which URL the MC should redirect to
+  // after successful login.
+  const applicationIdentifier = isProd
+    ? `${configurationData.id}:${entryPointUriPath}`
+    : `__local:${entryPointUriPath}`;
+
+  const developmentConfig = getRuntimeEnvironmentConfigForDevelopment({
+    isProd,
+    configurationData,
+    mcApiUrl,
+    appConfig,
+    entryPointUriPath,
+  });
+
+  return {
+    // Common config
+    ...omitEmpty<Record<string, unknown>>(additionalAppEnv),
+    cdnUrl: cdnUrl.href,
+    env: appEnvKey,
+    frontendHost: appUrl.host,
+    location: appConfig.cloudIdentifier,
+    mcApiUrl: mcApiUrl.origin,
+    revision,
+    servedByProxy: isProd,
+
+    // Application config
+    applicationId: applicationIdentifier,
+    applicationIdentifier,
```

Comment:
Previously this field was named `applicationId` but I prefer now to name it `applicationIdentifier` to make it less confusing with the actual application ID.

For backwards compatibility we'll keep it around for now.